src/linear_congruential_random_generator.c

```diff
@@ -2,22 +2,24 @@
 
 #include "linear_congruential_random_generator.h"
 
+#include <stdint.h> 
+
 /* The routines below implements the linear congruential random
    number generator. In particular, the function
    "set_linear_congruential_generator_seed" sets the seed of the random
    generator, while the function "linear_congruential_random_generator"
    returns a pseudo-random real number in the range [0, 1.0]. */
 
 /* Variables and pointers declarations */
-int ISEED = 16.; /* Default value in case the user does not specify the seed */
+uint16_t ISEED = 16; /* Default value in case the user does not specify the seed */
+#define LINEAR_RAND_MAX pow(2, sizeof(ISEED) * 8)
+
 
-/* Set the seed for the random number generator */
 void set_linear_congruential_generator_seed(int num) { ISEED = num; }
 
-/* Linear congruential random number generator */
 lcrg_real linear_congruential_random_generator(void) {
-  ISEED = fmod(75. * ISEED + 74., 65537.);
-  return (ISEED / 65537.);
+  ISEED = 75 * ISEED + 74;
+  return ISEED * (1.0 / LINEAR_RAND_MAX);
```

Comment:
Maybe you could add a define for the inverse of LINEAR_RAND_MAX in order to do a multiplication and not a multiplication + a division

Reply:
Adding a "define" won't change anything to the final code. Defines are evaluated by the preprocessor and code that contains one is simply replaced by its expression. 

The 1 / LINEAR_RAND_MAX is evaluated at compile time and does not require a division. 
In fact, writing "return ISEED / LINEAR_RAND_MAX;" returns the same result on avr-gcc since the compiler can evaluate the inverse multiplicative without any loss of precision.

https://godbolt.org/z/6ncP9hr6Y

Reply:
I added the 1 / LINEAR_RAND_MAX rather than ISEED / LINEAR_RAND_MAX because the compiler only optimize the division in the second example if we enable optimizations (O1)